### PR TITLE
Implement connection screen in onboard flow

### PR DESCRIPTION
**Pull request by cp-ishita-g**
- [x] implement connection screen
- [x] create secondary button class
- [x] manage join space and invite screen design according to onboard flow and normal flow

test flow - it will show after login to a new user only

join_space create_space

### CODE REVIEW

**Comment by kaushiksaliya** [comment]
add bottom padding for next button in pickNameScreen.

**Comment by kaushiksaliya** [comment]
on create space when we select any space name from suggestion and click on next button then getting permission_denied error.

**Comment by kaushiksaliya** [comment]
on pick name screen when user click on system back button without adding name then do not show main screen. fix this flow.

**Comment by kaushiksaliya** [comment]
no need to show this join space title if user coming from onboard.

Screenshot from 2024-07-16 09-04-26

File: app/lib/ui/flow/space/join/join_space_screen.dart [hunk 1]
```diff
@@ -60,16 +63,25 @@ class _JoinSpaceState extends ConsumerState<JoinSpace> {
...
+            const SizedBox(height: 16),
+            Text(
+              context.l10n.join_space_get_code_from_space_creator_title,
+              style: AppTextStyle.subtitle1.copyWith(
+                color: context.colorScheme.textDisabled,
+              ),
+            )
+          ],
           const SizedBox(height: 40),
           _inviteCode(context),
```
[the remark this answers is not in the document]
Comment (cp-ishita-g) [reply app/lib/ui/flow/space/join/join_space_screen.dart:81]: it is upper case only can you please check again

[the remark this answers is not in the document]
Comment (kaushiksaliya) [reply app/lib/ui/flow/space/join/join_space_screen.dart:81]: check this.

Screenshot from 2024-07-22 10-35-58

File: app/ios/Runner/Info.plist [hunk 2]
```diff
@@ -70,5 +70,7 @@
    	<string>App requires permission to access photos</string>
    	<key>NSCameraUsageDescription</key>
    	<string>App requires permission to access camera</string>
+   	<key>ApiMapKey</key>
+    <string>ABCDFGHI</string>
```
Comment (cp-radhika-s) [inline app/ios/Runner/Info.plist:74]: Replace `<string>ABCDFGHI</string>` with `<string>YOUR MAP API KEY</string>`

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

File: app/ios/Runner/Info.plist (hunk 2 above)
Comment (cp-ishita-g) [reply app/ios/Runner/Info.plist:74]: it's it will be published

Comment (cp-radhika-s) [reply app/ios/Runner/Info.plist:74]: I'm not saying to add api key, just add this text `YOUR MAP API KEY`

**Review by kaushiksaliya — APPROVED** [review]

File: app/lib/ui/flow/auth/sign_in/sign_in_method_screen.dart [hunk 3]
```diff
@@ -100,13 +100,22 @@ class _SignInMethodScreenState extends ConsumerState<SignInMethodScreen> {
...
     if (mounted && (user?.first_name == null || user!.first_name!.isEmpty)) {
-      await AppRoute.pickName.push(context);
+      await AppRoute.pickName(user: user).push(context);
     }
 
-    if (mounted) AppRoute.home.go(context);
+    if (state.isNewUser && mounted) {
+      AppRoute.connection.go(context);
+    } else {
+      navigateToHome();
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/auth/sign_in/sign_in_method_screen.dart:113]: Don't we need to check for `mounted`?

File: app/lib/ui/flow/onboard/pick_name_screen.dart [hunk 4]
```diff
@@ -28,62 +29,46 @@ class _PickNameScreenState extends ConsumerState<PickNameScreen> {
...
-          Padding(
-            padding: const EdgeInsets.only(left: 16, right: 16, bottom: 30),
+          BottomStickyOverlay(
             child: PrimaryButton(
               context.l10n.common_next,
               enabled: state.enableBtn,
               progress: state.savingUser,
               onPressed: () {
-                _notifier.saveUser();
+                _notifier.saveUser(widget.user ?? _notifier.user!);
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/onboard/pick_name_screen.dart:71]: Let's not access notifier property directly, it should be through state only

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

**Review by cp-radhika-s — APPROVED** [review]

**Pull request merged**